projector-agent/build.gradle

```diff
@@ -74,6 +75,7 @@ if (serverTargetClasspath != null && serverClassToLaunch != null) {
       "-Dswing.bufferPerWindow=false",
       "-Dorg.jetbrains.projector.agent.path=${project.file("build/libs/${project.name}-${project.version}.jar")}",
       "-Dorg.jetbrains.projector.agent.classToLaunch=$serverClassToLaunch",
+      "-Djava.system.class.loader=org.jetbrains.projector.util.loading.ProjectorClassLoader",
```

Comment:
Well, I haven't expected that. Is there a way to avoid the need of specifying this extra parameter? I've tried running `:projector-plugin:runIde` and it works, seems like without this extra argument.

I'm afraid we can't add the requirement for this parameter easily since there are many places where we can't change it :(

We can of course change add it to all the documentation, Docker scripts, projector-demo, and so on. But how we can do this for IDEA itself and our plugin? It will require an extra setup step and also can fail working if we remove Projector since the classloader won't be available anymore.

Reply:
Oh, I see. We can avoid it by using reflection (that's how it already works for plugin). I'm not a fan of it, but seems it's lesser of two evils